### PR TITLE
ICS4: Simplify sendPacket interface

### PR DESCRIPTION
**Pull request by mpoke**
Closes https://github.com/cosmos/ibc/issues/708

Before merging update the calls to sendPacket throughout the standard, i.e., 

- [x] [ICS 20](https://github.com/cosmos/ibc/blob/master/spec/app/ics-020-fungible-token-transfer/README.md#packet-relay), which entails changing the signature of `createOutgoingPacket` (already inconsistent with ICS-4, see https://github.com/cosmos/ibc/issues/709)
- [x] [ICS27](https://github.com/cosmos/ibc/blob/master/spec/app/ics-027-interchain-accounts/README.md#sendtx), which entail updating `SendTx` (already inconsistent with ICS-4)
- [x] [ICS 29](https://github.com/cosmos/ibc/blob/master/spec/app/ics-029-fee-payment/README.md#embedded-applications-calling-into-ics-4)
- [x] [ICS 30](https://github.com/cosmos/ibc/blob/master/spec/app/ics-030-middleware/README.md)

### CODE REVIEW

**Comment by mpoke** [comment]
If the changes are approved, I'll create different PRs for handling the tasks in the description.

**Review by cwgoes — APPROVED** [review]
ACK, makes sense to me!

**Review by AdityaSripal — APPROVED** [review]

**Review by colin-axner — APPROVED** [review]
Nice improvement! :raised_hands:  

Will a changelog be added to the repo soon? I fear we will forget to apply some of these changes in ibc-go if they aren't logged somewhere

**Comment by mpoke** [comment]
> Will a changelog be added to the repo soon? I fear we will forget to apply some of these changes in ibc-go if they aren't logged somewhere

@colin-axner That's a good point. @AdityaSripal @crodriguezvega what do you think?

**Review by seantking — APPROVED** [review]

File: spec/core/ics-004-channel-and-packet-semantics/README.md [hunk 1]
```diff
@@ -513,51 +513,56 @@ Represented spatially, packet transit between two machines can be rendered as fo
...
-    channel = provableStore.get(channelPath(packet.sourcePort, packet.sourceChannel))
-
-    // optimistic sends are permitted once the handshake has started
+function sendPacket(
+  capability: CapabilityKey,
+  sourcePort: Identifier,
+  sourceChannel: Identifier,
+  timeoutHeight: Height,
+  timeoutTimestamp: uint64,
+  data: bytes) {
```
Comment (colin-axner) [inline spec/core/ics-004-channel-and-packet-semantics/README.md:537]: sorry I am just noticing this now, but I think the packet sequence should be returned

Comment (mpoke) [reply spec/core/ics-004-channel-and-packet-semantics/README.md:537]: @colin-axner Could you please open an issue?

Comment (colin-axner) [reply spec/core/ics-004-channel-and-packet-semantics/README.md:537]: #819

**Pull request merged**